core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -1023,6 +1023,12 @@ public ReplicationGroup getReplicationGroup() {
         }
 
         public ReplicaResponse(long localCheckpoint) {
+            /*
+             * A replica should always know its own local checkpoint so this should always be a valid sequence number or the pre-6.0 local
+             * checkpoint value when simulating responses to replication actions that pre-6.0 nodes are not aware of (e.g., the global
+             * checkpoint background sync, and the primary/replica resync).
+             */
+            assert localCheckpoint != SequenceNumbersService.UNASSIGNED_SEQ_NO;
```

Comment:
can you also add this assertion to `GlobalCheckpointTracker.updateLocalCheckpoint` (the private method)? I guess that some tests will start to fail as they are not correctly exercising the tracker.